<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="rheaprag" />

<meta name="date" content="2021-04-22" />

<title>creatinine_clearance</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ACCOuNT</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/rheaprag/ACCOuNT">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">creatinine_clearance</h1>
<h4 class="author">rheaprag</h4>
<h4 class="date">2021-04-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ACCOuNT/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomrheapragACCOuNTtree836e4fc0be86920681e109f7bc07352ad159c88btargetblank836e4fca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/rheaprag/ACCOuNT/tree/836e4fc0be86920681e109f7bc07352ad159c88b" target="_blank">836e4fc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomrheapragACCOuNTtree836e4fc0be86920681e109f7bc07352ad159c88btargetblank836e4fca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/rheaprag/ACCOuNT/tree/836e4fc0be86920681e109f7bc07352ad159c88b" target="_blank">836e4fc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/creatinine_clearance.Rmd</code>) and HTML (<code>docs/creatinine_clearance.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/836e4fc0be86920681e109f7bc07352ad159c88b/analysis/creatinine_clearance.Rmd" target="_blank">836e4fc</a>
</td>
<td>
rheaprag
</td>
<td>
2021-10-29
</td>
<td>
Publish updated files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/2e54d57789b9632b44e94892d47d7bb38fe9dfec/docs/creatinine_clearance.html" target="_blank">2e54d57</a>
</td>
<td>
rheaprag
</td>
<td>
2021-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/dc06567a3533aa8e3ddf2ced5183e08168af6d7f/analysis/creatinine_clearance.Rmd" target="_blank">dc06567</a>
</td>
<td>
rheaprag
</td>
<td>
2021-09-10
</td>
<td>
Publish updated files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/a46b009be514d24846ebaf81058366440a216468/docs/creatinine_clearance.html" target="_blank">a46b009</a>
</td>
<td>
rheaprag
</td>
<td>
2021-08-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/5df004f0a5e2f66768cbaa7c7736f8a65a4909ff/analysis/creatinine_clearance.Rmd" target="_blank">5df004f</a>
</td>
<td>
rheaprag
</td>
<td>
2021-08-16
</td>
<td>
Update CrCl with covariate analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/680c0ddd098a63bf6697a3e95d98b6be8c0dd9df/docs/creatinine_clearance.html" target="_blank">680c0dd</a>
</td>
<td>
rheaprag
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/542a4a304944961400031407bf8853d9e5599a26/analysis/creatinine_clearance.Rmd" target="_blank">542a4a3</a>
</td>
<td>
rheaprag
</td>
<td>
2021-08-03
</td>
<td>
Publish updated CrCl analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/96b42221bdba4e0762544df9b3ef4ba86e7b71ef/docs/creatinine_clearance.html" target="_blank">96b4222</a>
</td>
<td>
rheaprag
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/ec9f04ac3c545761a3f785100a70952159d60b01/analysis/creatinine_clearance.Rmd" target="_blank">ec9f04a</a>
</td>
<td>
rheaprag
</td>
<td>
2021-07-28
</td>
<td>
Publish incomplete analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/77dac55423ff86743164d4d8f9ae9b32ddf6903b/docs/creatinine_clearance.html" target="_blank">77dac55</a>
</td>
<td>
rheaprag
</td>
<td>
2021-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/5c901a3ae643ec008c087b006b553912505ab61f/analysis/creatinine_clearance.Rmd" target="_blank">5c901a3</a>
</td>
<td>
rheaprag
</td>
<td>
2021-05-25
</td>
<td>
Added citatiosn and fixed column numbers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/ACCOuNT/33b030e99d31f8974013e4226d07c160780bcaac/docs/creatinine_clearance.html" target="_blank">33b030e</a>
</td>
<td>
rheaprag
</td>
<td>
2021-05-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/a608eaa007a137a2b04ce33d780ba0dd9ee183dd/analysis/creatinine_clearance.Rmd" target="_blank">a608eaa</a>
</td>
<td>
rheaprag
</td>
<td>
2021-05-19
</td>
<td>
Update website and fix typos
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This analysis looks at creatinine clearance in the data.</p>
<p>Creatinine is a waste product produced in our body. Low creatinine levels indicate kidney damage while high levels indicate diminished kidney function.</p>
<p>This analysis uses the Clopidogrel and NOAC drug groups.</p>
<div id="extract-phenotype-data" class="section level3">
<h3>Extract Phenotype Data</h3>
<p>The <code>csv</code> files with matched IDs were used for this.</p>
<pre><code>#adapted from Ashley&#39;s code ACCOUNT/scripts/pheno/extract_important_pheno_data.R

#import necessary libraries
library(data.table)
library(dplyr)

#import csv files for clop &amp; noac
clop&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/account_clopidogrel_processed_matched_IDs_all.csv&quot;)
noac&lt;-fread(&quot;/home/rhea/account_wflow/phenos/noac/account_noac_processed_matched_IDs_all.csv&quot;)

#save ids and pheno columns only, remove unneeded columns so files are easier to work with
clop_small&lt;-clop[, c(1:9,206)]
noac_small&lt;-noac[, c(1:9,79)]

#output files
fwrite(clop_small, &quot;clop_CrCl_with_ids.txt&quot;, sep = &quot;\t&quot;, quote = F, na = NA)
fwrite(noac_small, &quot;noac_CrCl_with_ids.txt&quot;, sep = &quot;\t&quot;, quote = F, na = NA)
</code></pre>
</div>
<div id="merge-samples-file-with-phenotype-data" class="section level3">
<h3>Merge Samples File with Phenotype Data</h3>
<pre><code>#import necessary libraries
library(data.table)
library(dplyr)

#import samples &amp; pheno txt files for clop 
clop_samples&lt;-fread(&quot;/home/ashley/account/dosages/blacks/clop/samples_clop.txt&quot;, header = FALSE)
clop_phenos&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/clop_CrCl_with_ids.txt&quot;)

#merge two files
clop_cmbnd&lt;-left_join(clop_samples, clop_phenos)

#output file
fwrite(clop_cmbnd, &quot;clop_combnd_pheno_ids.txt&quot;, sep = &quot;\t&quot;, quote = F, na = NA)
</code></pre>
<p>Repeat this step to create combined phenotype file for NOAC.</p>
</div>
<div id="calculate-creatinine-clearance-for-clopidogril-data" class="section level3">
<h3>Calculate Creatinine Clearance for Clopidogril Data</h3>
<pre><code>#crcl calculation
#Est. Creatinine Clearance = [[140 - age(yr)]*weight(kg)]/[72*serum Cr(mg/dL)]
#multiply by 0.85 for women

library(data.table)
library(dplyr)

#import pheno file in for clop, has missing data values
clop&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/clop_combnd_pheno_ids.txt&quot;)

str(clop) #see structure of data.frame
#Note Height and Creatinine level are characters, need to convert to the numeric type

clop_age&lt;-clop$`Age at enrollment (with over 89 binning)`
clop_weight&lt;-as.numeric(clop$`Weight (kg)`) #as.numeric() converts MD&#39;s to NA, I&#39;m assuming MD=&quot;missing data&quot;?
clop_sCr&lt;-as.numeric(clop$`Creatinine level (mg/dL)`) #as.numeric() converts characters to numbers and MD&#39;s to NA, so we can do math

crcl &lt;- (140 - clop_age)*clop_weight / (72*clop_sCr)

#use mutate() to add crcl to clop data.frame and adjust if female with ifelse()
clop &lt;- mutate(clop, CrCl = ifelse(Gender==&#39;M&#39;,crcl,0.85*crcl))

fwrite(clop, &quot;clop_CrCl_pheno.txt&quot;, sep=&quot;\t&quot;, na=&quot;NA&quot;, quote=FALSE)</code></pre>
</div>
<div id="finishing-data-file" class="section level3">
<h3>Finishing Data File</h3>
<p>Now, a row needs to be removed from both the clop and noac files as the person switched drugs:</p>
<pre><code>library(data.table)
library(dplyr)

clop&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/clop_CrCl_pheno.txt&quot;)
clop_new&lt;-clop[-c(16),]
fwrite(clop_new, &quot;clop_CrCl_pheno_final.txt&quot;, sep = &quot;\t&quot;, quote = F, na = NA)

noac&lt;-fread(&quot;/home/rhea/account_wflow/phenos/noac/noac_combnd_pheno_ids.txt&quot;)
noac_new&lt;-noac[-c(6),] #remove duplicated data &amp; missing data

fwrite(noac_new, &quot;noac_CrCl_pheno_final.txt&quot;, sep = &quot;\t&quot;, quote = F, na = NA)</code></pre>
</div>
<div id="histogram" class="section level3">
<h3>Histogram</h3>
<p>The original histogram created showed that the data was skewed to the right.</p>
<p>In order to fix this, I first applied log transformation to the phenotype files:</p>
<pre><code>library(ggplot2)
library(dplyr)

#clop
clop&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/clop_CrCl_pheno_final.txt&quot;)
Creatinine_clop &lt;- clop$CrCl
Creatinine_c_log &lt;- log(Creatinine_clop)
clop_hist &lt;- hist(Creatinine_c_log,
main=&quot;Creatinine Clearance of Clopidogril&quot;,
col=&quot;darkmagenta&quot;)

#noac 
noac&lt;-fread(&quot;/home/rhea/account_wflow/phenos/noac/noac_CrCl_pheno_final.txt&quot;)
colnames(noac)[10] &lt;- &quot;CrCl&quot;
Creatinine_noac &lt;- noac$CrCl
Creatinine_noac_num &lt;- as.numeric(Creatinine_noac)
Creatinine_noac_log &lt;- log(Creatinine_noac_num)
noac_hist &lt;- hist(Creatinine_noac_log,
main=&quot;Creatinine Clearance of NOAC drugs&quot;,
col=&quot;gold&quot;)
</code></pre>
<p>However, the log transformed data was now skewed to the left. So I applied rank normalization to the data instead.</p>
<pre><code>library(ggplot2)
library(dplyr)

#rank normalize files
source(&quot;/home/rhea/account_wflow/rntransform.R&quot;)
source(&quot;/home/rhea/account_wflow/ztransform.R&quot;)

clop&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop/clop_CrCl_pheno_final.txt&quot;)
Creatinine_clop &lt;- rntransform(clop$CrCl)
clop_hist &lt;- hist(Creatinine_clop,
main=&quot;Creatinine Clearance of Clopidogril&quot;,
col=&quot;darkmagenta&quot;)

noac&lt;-fread(&quot;/home/rhea/account_wflow/phenos/noac/noac_CrCl_pheno_final.txt&quot;)
colnames(noac)[10] &lt;- &quot;CrCl&quot;
Creatinine_noac &lt;- noac$CrCl
Creatinine_noac_num &lt;- as.numeric(Creatinine_noac)
Creatinine_noac_rn &lt;- rntransform(Creatinine_noac_num)
noac_hist &lt;- hist(Creatinine_noac_rn,
main=&quot;Creatinine Clearance of NOAC drugs&quot;,
col=&quot;gold&quot;)
</code></pre>
<p>These histograms showed a normal distribution. The phenotype data used for the remainder of the calculations will be rank-normalized.</p>
<p>Histograms are all saved under <code>/home/rhea/account_wflow/phenos/{drug}</code>.</p>
</div>
<div id="merge-samples-file--" class="section level3">
<h3>Merge samples file - ?</h3>
<pre><code>
clop_ids &lt;- fread(&quot;/home/rhea/account_wflow/vcfs/blacks_imputed/clop/clop_ids_CrCl.txt&quot;)
noac_ids &lt;- fread(&quot;/home/rhea/account_wflow/vcfs/blacks_imputed/noac/noac_ids_CrCl.txt&quot;)

clop_noac_ids &lt;- union_all(clop_ids, noac_ids)
fwrite(clop_noac_ids, &quot;clop_noac_ids.txt&quot;)
</code></pre>
<p>using merged samples file…</p>
</div>
<div id="create-dosage-files" class="section level3">
<h3>Create Dosage files</h3>
<pre><code>#split vcf by pulling out clop &amp; noac data using combined ID file
for chr in {1..22};
do vcftools --gzvcf /home/ashley/account/vcfs/blacks_imputed/all/chr${chr}_maf01_r28.vcf.gz --keep /home/rhea/account_wflow/vcfs/blacks_imputed/clop_noac/clop_noac_CrCl_ids.txt --recode --recode-INFO-all --out clop_noac_chr${chr};
done

#gzip new vcf files
gzip clop_noac_chr*.recode.vcf

# Use vcf_to_dosage.py to output dosage txt files
for chr in {1..22};
do python /home/rhea/account_wflow/vcfs/vcf_to_dosage.py -i /home/rhea/account_wflow/vcfs/blacks_imputed/clop_noac/ -c ${chr} -d clop_noac;
done

gzip clop_noacchr*.dosage.txt
</code></pre>
<p>create merged samples_clop_noac.txt file</p>
</div>
<div id="run-pwas" class="section level3">
<h3>Run PWAS</h3>
<pre><code>
#Run PrediXcan with drug groups and protein models:

declare -a drugs=(&quot;clop_noac&quot;)
declare -a model_pops=(&quot;AFA&quot; &quot;ALL&quot; &quot;CAU&quot; &quot;CHN&quot; &quot;HIS&quot;)

for model_pop in ${model_pops[@]}
  do
  for drug in ${drugs[@]}
    do
    python3 /home/wheelerlab3/MetaXcan/software/Predict.py \
    --model_db_path /home/ryan/TOPMed_Proteome/dbs_out/${model_pop}_PCAIR_baseline_models_PAV_filtered_rho0.1_zpval0.05.db \
    --text_genotypes /home/rhea/account_wflow/dosages/blacks/clop_noac/pred_dosages/clop_noacchr*.dosage.txt.gz \
    --text_sample_ids /home/rhea/account_wflow/dosages/blacks/clop_noac/samples_clop_noac.txt \
    --prediction_output /home/rhea/account_wflow/pwas_results/blacks/${drug}/predicted_levels/blacks_${drug}_PCAIR_PAV_filtered_${model_pop}_baseline_rho0.1_zpval0.05.txt
  done
done
</code></pre>
</div>
<div id="convert-predixcan-output-to-gemma-input" class="section level3">
<h3>Convert PrediXcan Output to GEMMA input</h3>
<pre><code>
library(data.table)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
model_list &lt;- c(&quot;AFA&quot;, &quot;HIS&quot;, &quot;CHN&quot;, &quot;ALL&quot;)
drug_list &lt;- c(&quot;clop_noac&quot;)

for(model in model_list){
  for(drug in drug_list){
    pred_exp &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/predicted_levels/blacks_&quot; %&amp;% drug %&amp;% &quot;_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.txt&quot;, header = F) #path to PrediXcan-produced _predicted_expression.txt
    pred_exp$V1 &lt;- NULL
    pred_exp$V2 &lt;- NULL
    pred_exp &lt;- transpose(pred_exp)
    gene_list &lt;- as.data.frame(pred_exp$V1)
    colnames(gene_list) &lt;- &quot;V0&quot;
    pred_exp &lt;- cbind(gene_list, pred_exp)
    pred_exp$V0 &lt;- NA
    pred_exp$V1 &lt;- NA
    colnames(gene_list) &lt;- &quot;V&quot;
    pred_exp &lt;- cbind(gene_list, pred_exp)
    fwrite(pred_exp, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/assoc_gemma_input/blacks_&quot; %&amp;% drug %&amp;% &quot;_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05_gemma.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = F) #path to GEMMA input/pseudo-genotype
  }
}</code></pre>
</div>
<div id="run-covariates" class="section level3">
<h3>Run Covariates</h3>
<p>Check <code>Covariates</code> on Home. This is where the covariates file, phenotype file, and kinship files will be generated.</p>
</div>
<div id="run-gemma" class="section level3">
<h3>Run GEMMA</h3>
<pre><code>
declare -a drugs=(&quot;clop_noac&quot;)
declare -a models=(&quot;ALL&quot; &quot;AFA&quot; &quot;CHN&quot; &quot;HIS&quot;)

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for drug in ${drugs[@]}
  do
  for model in ${models[@]}
    do
    /usr/local/bin/gemma -g /home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_input/blacks_clop_noac_PCAIR_PAV_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
    -notsnp \
    -p /home/rhea/account_wflow/phenos/clop_noac/clop_noac_pheno_no_ids.txt \
    -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
    -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
    -lmm 4 \
    -o ${drug}_PCAIR_PAV_filtered_${model}_baseline_rho0.1_zpval0.05
  done
done
</code></pre>
</div>
<div id="correct-bonferroni" class="section level3">
<h3>Correct Bonferroni</h3>
<pre><code>#Import necessary libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in drug name
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

#Create list for file input variants
drug_list &lt;- c(&quot;clop_noac&quot;)
model_list &lt;- c(&quot;ALL&quot;, &quot;AFA&quot;, &quot;CHN&quot;, &quot;HIS&quot;) 

#Make a data frame from combined PrediXcan results that includes chr #, bp, and BH and Bonferroni adj p-value
  #Read in output files
  #Create vector of p-values
  #Adjust p-values with BH and bonferroni methods
  #Add new columns with adjusted p-values to data frame
  #Join PrediXcan data frame with bp_chrome data frame to add columns with chr # and bp
  #Remove old/inaccurate chr and bp columns
  #Ouput data frame into directory
for (drug in drug_list) {
  for (model in model_list) {
    assoc_file &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/assoc_gemma_output/&quot; %&amp;% drug %&amp;% &quot;_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.assoc.txt&quot;)
    pvalues &lt;- select(assoc_file, contains(&quot;p_wald&quot;))
    pvalues &lt;- as.vector(unlist(pvalues))
    pvalues_adjusted_BH &lt;- p.adjust(pvalues, method = &quot;BH&quot;)
    pvalues_adjusted_BF &lt;- p.adjust(pvalues, method = &quot;bonferroni&quot;)
    assoc_file &lt;- add_column(assoc_file,  pvalues_adjusted_BH = pvalues_adjusted_BH , .before = &quot;p_lrt&quot;)
    assoc_file &lt;- add_column(assoc_file,  pvalues_adjusted_bonferroni = pvalues_adjusted_BF, .before = &quot;p_lrt&quot;)
    fwrite(assoc_file, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.adj.txt&quot;, sep = &quot;\t&quot;, quote = F)
  }
}</code></pre>
</div>
<div id="combine-significant-hits" class="section level3">
<h3>Combine Significant Hits</h3>
<pre><code>#Import necessasry libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in model name
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

#Create list of models for file input
model_list &lt;- c(&quot;ALL&quot;, &quot;AFA&quot;)
drug_list &lt;- c(&quot;clop_noac&quot;)

#Make a data frame with all results from each model per drug
  #Read in file
  #Add column containing model name
  #Compile significant subsets into single data frame
for(drug in drug_list){
  for(model in model_list){
    output &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.adj.txt&quot;)
    output &lt;- add_column(output, model = model, .before = &quot;chr&quot;)
    output &lt;- add_column(output, drug = drug, .before = &quot;chr&quot;)
    if(exists(&quot;all_assoc&quot;)){
      all_assoc&lt;-merge(x = all_assoc, y = output, all = TRUE)
    }
    else{
      all_assoc&lt;-output
    }
  }
  
  #subset at different thresholds
  all_sig&lt;-subset(all_assoc, pvalues_adjusted_BH &lt; .75)
  most_sig&lt;-subset(all_assoc, pvalues_adjusted_BH &lt; .1)
  BF_not1&lt;-subset(all_assoc, pvalues_adjusted_bonferroni &lt; 1)
  BF_sig&lt;-subset(all_assoc, pvalues_adjusted_bonferroni &lt; .05)
   
  #Output data frames into directory
  fwrite(all_assoc, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_rn_allassoc_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(all_sig, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_rn_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(most_sig, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_rn_most_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(BF_not1, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_rn_BFnot1_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(BF_sig, &quot;/home/rhea/account_wflow/pwas_results/blacks/&quot; %&amp;% drug %&amp;% &quot;/adj_assoc_output/&quot; %&amp;% drug %&amp;% &quot;_rn_BFsig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
}</code></pre>
</div>
<div id="create-plots" class="section level3">
<h3>Create Plots</h3>
<pre><code>library(data.table)
library(dplyr)
library(ggplot2)

#PrediXcan PWAS Results:

#Pheno files:
clop_noac_pheno&lt;-fread(&quot;/home/rhea/account_wflow/phenos/clop_noac/clop_noac_rn_pheno_no_ids.txt&quot;, header = F)
clop_noac_samples&lt;-fread(&quot;/home/rhea/account_wflow/dosages/blacks/clop_noac/samples_clop_noac.txt&quot;, header = F)

x_clop_noac&lt;-cbind(clop_noac_samples, clop_noac_pheno)
colnames(x_clop_noac)&lt;-c(&quot;FID&quot;, &quot;IID&quot;, &quot;pheno&quot;)
 
#Predicted Expression files:
y_clop_noac_Pred_AFA&lt;-fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/clop_noac/predicted_levels/blacks_clop_noac_PCAIR_PAV_filtered_AFA_baseline_rho0.1_zpval0.05.txt&quot;)


#Plot Predicted Expression for IL5RA, for AFA Model, ALL pop, and Ara-C drug: *******???????********
#Data file:
xy_clop_noac_Pred_AFA&lt;-inner_join(x_clop_noac, y_clop_noac_Pred_AFA, by = c(&quot;FID&quot;, &quot;IID&quot;))

******
ggplot(xy_clop_noac_Pred_AFA, aes(x = xy_clop_noac_Pred_AFA$SL006512_ENSG00000109758.8, y = xy_clop_noac_Pred_AFA$pheno)) +
  geom_jitter(size = 0.75, color = &quot;#ec328c&quot;) + 
  geom_density_2d(color = &quot;#ffbf24&quot;) + 
  stat_smooth(method=&quot;lm&quot;, se = TRUE, fullrange = TRUE, color = &quot;#8c1788&quot;) + 
  scale_x_continuous(name = &quot;Predicted protein levels&quot;) + 
  scale_y_continuous(name = &quot;Clopidogril_NOAC Dose (rank normalized)&quot;) + 
  theme_bw() + 
  theme(text = element_text(size = 12), plot.title = element_text(hjust = 0.5)) +
  ggtitle(&quot;Clopidogril_NOAC Baseline AFA AFM&quot;)
dev.off()</code></pre>
</div>
<div id="create-qq-manhattan-plots" class="section level3">
<h3>Create qq &amp; manhattan plots</h3>
<pre><code>#Import necessary libraries
library(data.table)
library(dplyr)
library(qqman)
library(colorspace)

#Create function to paste in drug name
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

#Create color vector
colors&lt;-sequential_hcl(4,&quot;SunsetDark&quot;)

#Create list of drugs for file input
drug_list &lt;- c(&quot;clop_noac&quot;)
model_list &lt;- c(&quot;AFA&quot;, &quot;ALL&quot;)

#Make QQ and Manhattan plots using loop
for(drug in drug_list){
  for(model in model_list){
    pwas_output &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_output/clop_noac_PCAIR_PAV_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.assoc.txt&quot;)
    anno &lt;- fread(&quot;/home/ryan/topmed/proteome/proteome_with_multigene_expression/annotation_all_aptamers_ENSG.txt&quot;)
    pwas_out_anno&lt;-left_join(pwas_output, anno, by = c(&quot;rs&quot; = &quot;gene_id&quot;))
    pwas_out_anno$chr.y&lt;-as.numeric(pwas_out_anno$chr.y)
    
    png(filename = &quot;pwas_clop_noac_&quot; %&amp;% model %&amp;% &quot;.qqplot.png&quot;, res=100)
    qq(pwas_out_anno$p_wald)
    dev.off()

    png(filename = &quot;pwas_clop_noac_&quot; %&amp;% model %&amp;% &quot;.manplot.png&quot;, res=100)
    manhattan(pwas_out_anno, chr = &quot;chr.y&quot;, bp = &quot;start&quot;, p = &quot;p_wald&quot;, col = colors)
    dev.off()
  }
}

pwas_output_ALL &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_output/clop_noac_PCAIR_PAV_filtered_ALL_baseline_rho0.1_zpval0.05.assoc.txt&quot;)
pwas_out_anno_ALL&lt;-left_join(pwas_output_ALL, anno, by = c(&quot;rs&quot; = &quot;gene_id&quot;))
pwas_out_anno_ALL$chr.y&lt;-as.numeric(pwas_out_anno_ALL$chr.y)
qq(pwas_out_anno_ALL$p_wald)
manhattan(pwas_out_anno_ALL, chr = &quot;chr.y&quot;, bp = &quot;start&quot;, p = &quot;p_wald&quot;, snp = &quot;rs&quot;, col = colors)

pwas_output_AFA &lt;- fread(&quot;/home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_output/clop_noac_PCAIR_PAV_filtered_ALL_baseline_rho0.1_zpval0.05.assoc.txt&quot;)
pwas_out_anno_AFA&lt;-left_join(pwas_output_AFA, anno, by = c(&quot;rs&quot; = &quot;gene_id&quot;))
pwas_out_anno_AFA$chr.y&lt;-as.numeric(pwas_out_anno_AFA$chr.y)
qq(pwas_out_anno_AFA$p_wald)
manhattan(pwas_out_anno_AFA, chr = &quot;chr.y&quot;, bp = &quot;start&quot;, p = &quot;p_wald&quot;, snp = &quot;rs&quot;, col = colors)
</code></pre>
</div>
<div id="gwas---gemma" class="section level3">
<h3>GWAS - GEMMA</h3>
<pre><code>
declare -a drugs=(&quot;clop_noac&quot;)
declare -a models=(&quot;ALL&quot; &quot;AFA&quot; &quot;CHN&quot; &quot;HIS&quot;)

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for drug in ${drugs[@]}
  do
  for model in ${models[@]}
    do
    /usr/local/bin/gemma -g /home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_input/blacks_clop_noac_PCAIR_PAV_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
    -notsnp \
    -g /home/rhea/account_wflow/**/clop_noac/clop_noac_pheno_no_ids.txt \
    -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
    -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
    -lmm 4 \
    -o ${drug}_PCAIR_PAV_filtered_${model}_baseline_rho0.1_zpval0.05
  done
done
</code></pre>
<p>```</p>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
