<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="rheaprag" />

<meta name="date" content="2022-03-21" />

<title>twas</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ACCOuNT</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/rheaprag/ACCOuNT">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">twas</h1>
<h4 class="author">rheaprag</h4>
<h4 class="date">2022-03-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ACCOuNT/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomrheapragACCOuNTtree4ab6f195a914003f3fd9ed270128215effe3d509targetblank4ab6f19a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/rheaprag/ACCOuNT/tree/4ab6f195a914003f3fd9ed270128215effe3d509" target="_blank">4ab6f19</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomrheapragACCOuNTtree4ab6f195a914003f3fd9ed270128215effe3d509targetblank4ab6f19a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/rheaprag/ACCOuNT/tree/4ab6f195a914003f3fd9ed270128215effe3d509" target="_blank">4ab6f19</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/gwas_plots.Rmd
    Untracked:  clop_CrCl_pheno.txt

Unstaged changes:
    Modified:   analysis/calculation_clop_CrCl.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/twas.Rmd</code>) and HTML (<code>docs/twas.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/ACCOuNT/blob/4ab6f195a914003f3fd9ed270128215effe3d509/analysis/twas.Rmd" target="_blank">4ab6f19</a>
</td>
<td>
rheaprag
</td>
<td>
2022-04-04
</td>
<td>
Publish TWAS and remove unneccesary analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This details the script for running a TWAS on the ACCOuNT data.</p>
</div>
<div id="run-predixcan-with-transcriptome-models" class="section level2">
<h2>Run PrediXcan with transcriptome models</h2>
<pre><code>#Run PrediXcan with drug groups and protein models:

#--elasticnet_model_db_path /home/daniel/mashr/EN/WGS_files/models/PBMC_${model_pop}_PCAIR_elasticnet_models_rho0.1_zpval0.05.db \
#--mashr_model_db_path /home/daniel/mashr/mashr_db/WGS_files/mashr_models/... \


declare -a drugs=(&quot;clop_noac&quot;)
declare -a model_pops=(&quot;AFA&quot; &quot;ALL&quot;)

for model_pop in ${model_pops[@]}
  do
  for drug in ${drugs[@]}
    do
    python3 /home/wheelerlab3/MetaXcan/software/Predict.py \
    --model_db_path /home/daniel/mashr/EN/WGS_files/models/PBMC_${model_pop}_PCAIR_elasticnet_models_rho0.1_zpval0.05.db \
    --text_genotypes /home/rhea/account_wflow/dosages/blacks/clop_noac/pred_dosages/clop_noacchr*.dosage.txt.gz \
    --text_sample_ids /home/rhea/account_wflow/dosages/blacks/clop_noac/samples_clop_noac.txt \
    --prediction_output /home/rhea/account_wflow/twas_results/predicted_levels/clop_noac_CrCl_PCAIR_filtered_${model_pop}_baseline_rho0.1_zpval0.05.txt
  done
done</code></pre>
</div>
<div id="convert-predixcan-output-to-gemma-input" class="section level2">
<h2>Convert PrediXcan output to GEMMA input</h2>
<pre><code>library(data.table)
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)
model_list &lt;- c(&quot;AFA&quot;, &quot;ALL&quot;)

for(model in model_list){
  pred_exp &lt;- fread(&quot;/home/rhea/account_wflow/twas_results/predicted_levels/clop_noac_CrCl_PCAIR_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.txt&quot;, header = F) #path to PrediXcan-produced _predicted_expression.txt
  pred_exp$V1 &lt;- NULL
  pred_exp$V2 &lt;- NULL
  pred_exp &lt;- transpose(pred_exp)
  gene_list &lt;- as.data.frame(pred_exp$V1)
  colnames(gene_list) &lt;- &quot;V0&quot;
  pred_exp &lt;- cbind(gene_list, pred_exp)
  pred_exp$V0 &lt;- NA
  pred_exp$V1 &lt;- NA
  colnames(gene_list) &lt;- &quot;V&quot;
  pred_exp &lt;- cbind(gene_list, pred_exp)
  fwrite(pred_exp, &quot;/home/rhea/account_wflow/twas_results/assoc_gemma_input/clop_noac_CrCl_PCAIR_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05_gemma.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = F) #path to GEMMA input/pseudo-genotype
}</code></pre>
</div>
<div id="gemma-analysis" class="section level2">
<h2>GEMMA analysis</h2>
<pre><code>declare -a models=(&quot;ALL&quot; &quot;AFA&quot;)

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for model in ${models[@]}
  do
  /usr/local/bin/gemma -g /home/rhea/account_wflow/twas_results/assoc_gemma_input/clop_noac_CrCl_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
  -notsnp \
  -p /home/rhea/account_wflow/phenos/clop_noac/clop_noac_rn_pheno_no_ids.txt \
  -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
  -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
  -lmm 4 \
  -o clop_noac_CrCl_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05
done


#new models
declare -a drugs=(&quot;clop_noac&quot;)
declare -a models=(&quot;ALL&quot; &quot;AFA&quot; &quot;CHN&quot; &quot;HIS&quot;)

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for drug in ${drugs[@]}
  do
  for model in ${models[@]}
    do
    /usr/local/bin/gemma -g /home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_input/blacks_clop_noac_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
    -notsnp \
    -p /home/rhea/account_wflow/phenos/clop_noac/clop_noac_rn_pheno_no_ids.txt \
    -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
    -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
    -lmm 4 \
    -o ${drug}_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05
  done
done</code></pre>
</div>
<div id="correct-bonferroni" class="section level2">
<h2>Correct Bonferroni</h2>
<pre><code>#Import necessary libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in drug name
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

#Create list for file input variants
model_list &lt;- c(&quot;ALL&quot;, &quot;AFA&quot;) 

#Make a data frame from combined PrediXcan results that includes chr #, bp, and BH and Bonferroni adj p-value
  #Read in output files
  #Create vector of p-values
  #Adjust p-values with BH and bonferroni methods
  #Add new columns with adjusted p-values to data frame
  #Join PrediXcan data frame with bp_chrome data frame to add columns with chr # and bp
  #Remove old/inaccurate chr and bp columns
  #Ouput data frame into directory

for (model in model_list) {
  assoc_file &lt;- fread(&quot;/home/rhea/account_wflow/twas_results/assoc_gemma_output/clop_noac_CrCl_PCAIR_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.assoc.txt&quot;)
  pvalues &lt;- select(assoc_file, contains(&quot;p_wald&quot;))
  pvalues &lt;- as.vector(unlist(pvalues))
  pvalues_adjusted_BH &lt;- p.adjust(pvalues, method = &quot;BH&quot;)
  pvalues_adjusted_BF &lt;- p.adjust(pvalues, method = &quot;bonferroni&quot;)
  assoc_file &lt;- add_column(assoc_file,  pvalues_adjusted_BH = pvalues_adjusted_BH , .before = &quot;p_lrt&quot;)
  assoc_file &lt;- add_column(assoc_file,  pvalues_adjusted_bonferroni = pvalues_adjusted_BF, .before = &quot;p_lrt&quot;)
  fwrite(assoc_file, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_PCAIR_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.adj.txt&quot;, sep = &quot;\t&quot;, quote = F)
}</code></pre>
<div id="combine-significant-hits" class="section level3">
<h3>Combine Significant Hits</h3>
<pre><code>#Import necessasry libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in model name
&quot;%&amp;%&quot; = function(a,b) paste(a,b,sep=&quot;&quot;)

#Create list of models for file input
model_list &lt;- c(&quot;ALL&quot;, &quot;AFA&quot;)

#Make a data frame with all results from each model per drug
  #Read in file
  #Add column containing model name
  #Compile significant subsets into single data frame
for(model in model_list){
  output &lt;- fread(&quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_PCAIR_filtered_&quot; %&amp;% model %&amp;% &quot;_baseline_rho0.1_zpval0.05.adj.txt&quot;)
  output &lt;- add_column(output, model = model, .before = &quot;chr&quot;)
  output &lt;- add_column(output, drug = drug, .before = &quot;chr&quot;)
  if(exists(&quot;all_assoc&quot;)){
      all_assoc&lt;-left_join(x = all_assoc, y = output, all = TRUE)
  }
  else{
    all_assoc&lt;-output
  }

  #subset at different thresholds
  all_sig&lt;-subset(all_assoc, pvalues_adjusted_BH &lt; .75)
  most_sig&lt;-subset(all_assoc, pvalues_adjusted_BH &lt; .1)
  BF_not1&lt;-subset(all_assoc, pvalues_adjusted_bonferroni &lt; 1)
  BF_sig&lt;-subset(all_assoc, pvalues_adjusted_bonferroni &lt; .05)
   
  #Output data frames into directory
  fwrite(all_assoc, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_allassoc_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(all_sig, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(most_sig, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_most_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(BF_not1, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_BFnot1_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
  fwrite(BF_sig, &quot;/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_BFsig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt&quot;, na = &quot;NA&quot;, quote = F, sep = &quot;\t&quot;, col.names = T) 
}</code></pre>
<p>No significant genes were found in this analysis.</p>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
