---
title: "gwas_plots"
author: "rheaprag"
date: "2022-03-21"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction
While there were no results from the GWAS, QQ and Manhattan plots were created. 

#QQ Plot
```{}
#Import necessary libraries
library(data.table)
library(dplyr)
library(qqman)
library(colorspace)

#Create color vector
colors<-sequential_hcl(4,"SunsetDark")

anno <- fread("/home/ryan/topmed/proteome/proteome_with_multigene_expression/annotation_all_aptamers_ENSG.txt")

#Make QQ Plots
gwas_output<- fread("/home/rhea/account_wflow/gwas_results/blacks/clop_noac/clop_noac_CrCl_gwas.assoc.txt.gz")
gwas_out_anno<-left_join(gwas_output, anno, by = c("rs" = "gene_id"))
gwas_out_anno$chr.y<-as.numeric(gwas_out_anno$chr.y)
qq(gwas_out_anno$p_wald)


sort all pvalues, figure out top 10-20

gwas_sorted <- gwas_out_anno[order(p_wald),]

dplyr filter all p values , keep less than .01

png(filename = "gwas_CrCl".qqplot.png", res=100)
qq(gwas_out_anno$p_wald)
dev.off()

use locus zoom of top 1 or 2 hits, see if there are any kidney related genes
- download summary stats and upload to locus zoom server
```

```{r, echo = FALSE}
#Import necessary libraries
library(data.table)
library(dplyr)
library(qqman)
library(colorspace)

#Create color vector
colors<-sequential_hcl(4,"SunsetDark")

anno <- fread("/home/ryan/topmed/proteome/proteome_with_multigene_expression/annotation_all_aptamers_ENSG.txt")

#Make QQ Plots
gwas_output<- fread("/home/rhea/account_wflow/gwas_results/blacks/clop_noac/clop_noac_CrCl_gwas.assoc.txt.gz")
gwas_out_anno<-left_join(gwas_output, anno, by = c("rs" = "gene_id"))
gwas_out_anno$chr.y<-as.numeric(gwas_out_anno$chr.y)
qq(gwas_out_anno$p_wald)
```

##Manhattan Plot
```{}
#import necessary libraries
library(tidyverse)
library(data.table)
library(qqman)
library(colorspace)
library(stringr)

colors<-sequential_hcl(4,"SunsetDark")

gwas_results <- fread("/home/rhea/account_wflow/gwas_results/blacks/clop_noac/adj_assoc_output/clop_noac_CrCl_gwas.adj.txt",stringsAsFactors=F)

gwas_pscore <- gwas_results %>% filter(p_score <= 0.01)

fwrite(gwas_pscore, "/home/rhea/account_wflow/gwas_results/clop_noac_CrCl_gwas_keep.txt", na = "NA", quote = F, sep = "\t", col.names = T)

#then split with python script
gwas_keep <- fread("/home/rhea/account_wflow/gwas_results/clop_noac_CrCl_gwas_keep.txt")

manhattan(gwas_keep, chr = "chr", bp = "start", p = "p_wald", snp = "ps", col = colors)

table(gwas_keep$chr)

```

## LocusZoom
```{}
#gwas_extract_columns.py

#pull out important columns from gwas_results
gwas_keep = pd.read_csv("/home/rhea/account_wflow/gwas_results/blacks/clop_noac/clop_noac_allassoc_CrCl_gwas.adj.txt", sep = "\t")

for idx,column in enumerate(gwas_keep.columns):
  print(idx,column)
#important columns to keep
  #chr - 1
  #allele1 - 4
  #allele0 - 5
  #start - 17
  #p_wald - 12
  #beta - 7
  #se - 8

data = gwas_keep.iloc[:,[1, 4, 5, 17, 12, 7, 8]]

data.to_csv("/home/rhea/account_wflow/gwas_results/clop_noac_CrCl_gwas_all.txt")
```

I used the `gwas_extract_columns.py` script to pull out important columns. This will allow for all of the GWAS results to be used for LocusZoom. 

Results can be found [here](https://my.locuszoom.org/gwas/24901/).
