---
title: "pred_dosages"
author: "rheaprag"
date: "2021-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In order to properly analyze the data in following steps, the original VCF files converted to dosage files and then converted to dosage files.

### Merge samples file

The IDs for clop and noac need to be combined for the combined dosage files.

```{}
clop_ids <- fread("/home/rhea/account_wflow/vcfs/blacks_imputed/clop/clop_ids_CrCl.txt")
noac_ids <- fread("/home/rhea/account_wflow/vcfs/blacks_imputed/noac/noac_ids_CrCl.txt")

clop_noac_ids <- union_all(clop_ids, noac_ids)
fwrite(clop_noac_ids, "clop_noac_ids.txt")

``` 

## VCF to Dosage Script

A Python script was then written in order to output the dosage files: 

```{}
import gzip
import re
import sys
import argparse
import os

def check_arg(args=None):
    parser = argparse.ArgumentParser(description='Script to filter imputed VCF')
    parser.add_argument('-i', '--inputdir',
                        help='directory containing VCF file',
                        required='True'
                        )
    parser.add_argument('-c', '--chr',
                        help='chromosome',
			type=str,
                        required='True'
                        )
     parser.add_argument('-d', '--drug',
                        help='drug',
			type=str,
                        required='True'
                        )
    return parser.parse_args(args)

#retrieve command line arguments
args = check_arg(sys.argv[1:])
chrpath = args.inputdir
c = args.chr
drug = args.drug
chrfile = chrpath + drug + "_" + "chr" + c + ".recode.vcf.gz"
mypath = "/home/rhea/account_wflow/"

# get dosage file data - getting errors with this section
if(os.path.exists(mypath+ 'UMich_dosages/') == False): 
    os.mkdir(mypath + 'UMich_dosages/')

outdosage = open(mypath + "UMich_dosages/" + drug + "chr" + c + ".dosage.txt","w")

for line in gzip.open(chrfile):
    #skip meta data and header lines
    if(line.startswith(b'##')):
        continue
    if(line.startswith(b'#CHROM')):
        continue
    arr = line.strip().split()
    (chr, pos, id, ref, alt, qual, filter, info, format) = arr[0:9]
    gt_dosagerow = arr[9:]
    #get freq of alt allele:
    info_list = info.split(b";")
    af_list = info_list[0].split(b"=")
    freqalt = af_list[1]
    #get dosage of alt allele
    dosagerow = map(lambda x : float(x.split(b":")[1]), gt_dosagerow) #lambda function to split each info entry and collect the dosage
    dosages = ' '.join(map(str,dosagerow))
    #reformat ids for proteome models (chr:pos)
    id_list = id.split(b":")
    id_formatted = str(id_list[0]) + ":" + str(id_list[1])
    output = str(chr) + ' ' + str(id_formatted) + ' ' + str(pos) + ' ' + str(ref) + ' ' + str(alt) + ' ' + str(freqalt) + ' ' + dosages + '\n'
    output = output.replace("b", "")
    output = output.replace("'", "")
    outdosage.write(output)

outdosage.close()
```

### Create Dosage files

The clop and noac data is pulled out of the original VCF file using the combined ID file made above. A loop then uses `vcf_to_dosage.py` file is then applies the python script to the new VCF file and outputs the dosage txt fies.

```{}
#split vcf by pulling out clop & noac data using combined ID file
for chr in {1..22};
do vcftools --gzvcf /home/ashley/account/vcfs/blacks_imputed/all/chr${chr}_maf01_r28.vcf.gz --keep /home/rhea/account_wflow/vcfs/blacks_imputed/clop_noac/clop_noac_CrCl_ids.txt --recode --recode-INFO-all --out clop_noac_chr${chr};
done

#gzip new vcf files
gzip clop_noac_chr*.recode.vcf

# Use vcf_to_dosage.py to output dosage txt files
for chr in {1..22};
do python /home/rhea/account_wflow/vcfs/vcf_to_dosage.py -i /home/rhea/account_wflow/vcfs/blacks_imputed/clop_noac/ -c ${chr} -d clop_noac;
done

gzip clop_noacchr*.dosage.txt

```

Dosage files are saved under `/home/rhea/account_wflow/dosages/blacks/clop_noad/pred_dosages`.
