---
title: "twas"
author: "rheaprag"
date: "2022-03-21"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

This details the script for running a TWAS on the ACCOuNT data. 

## Run PrediXcan with transcriptome models

```{}
#Run PrediXcan with drug groups and protein models:

#--elasticnet_model_db_path /home/daniel/mashr/EN/WGS_files/models/PBMC_${model_pop}_PCAIR_elasticnet_models_rho0.1_zpval0.05.db \
#--mashr_model_db_path /home/daniel/mashr/mashr_db/WGS_files/mashr_models/... \


declare -a drugs=("clop_noac")
declare -a model_pops=("AFA" "ALL")

for model_pop in ${model_pops[@]}
  do
  for drug in ${drugs[@]}
    do
    python3 /home/wheelerlab3/MetaXcan/software/Predict.py \
    --model_db_path /home/daniel/mashr/EN/WGS_files/models/PBMC_${model_pop}_PCAIR_elasticnet_models_rho0.1_zpval0.05.db \
    --text_genotypes /home/rhea/account_wflow/dosages/blacks/clop_noac/pred_dosages/clop_noacchr*.dosage.txt.gz \
    --text_sample_ids /home/rhea/account_wflow/dosages/blacks/clop_noac/samples_clop_noac.txt \
    --prediction_output /home/rhea/account_wflow/twas_results/predicted_levels/clop_noac_CrCl_PCAIR_filtered_${model_pop}_baseline_rho0.1_zpval0.05.txt
  done
done
```

## Convert PrediXcan output to GEMMA input

``` {}
library(data.table)
"%&%" = function(a,b) paste(a,b,sep="")
model_list <- c("AFA", "ALL")

for(model in model_list){
  pred_exp <- fread("/home/rhea/account_wflow/twas_results/predicted_levels/clop_noac_CrCl_PCAIR_filtered_" %&% model %&% "_baseline_rho0.1_zpval0.05.txt", header = F) #path to PrediXcan-produced _predicted_expression.txt
  pred_exp$V1 <- NULL
  pred_exp$V2 <- NULL
  pred_exp <- transpose(pred_exp)
  gene_list <- as.data.frame(pred_exp$V1)
  colnames(gene_list) <- "V0"
  pred_exp <- cbind(gene_list, pred_exp)
  pred_exp$V0 <- NA
  pred_exp$V1 <- NA
  colnames(gene_list) <- "V"
  pred_exp <- cbind(gene_list, pred_exp)
  fwrite(pred_exp, "/home/rhea/account_wflow/twas_results/assoc_gemma_input/clop_noac_CrCl_PCAIR_filtered_" %&% model %&% "_baseline_rho0.1_zpval0.05_gemma.txt", na = "NA", quote = F, sep = "\t", col.names = F) #path to GEMMA input/pseudo-genotype
}
```

## GEMMA analysis

```{}
declare -a models=("ALL" "AFA")

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for model in ${models[@]}
  do
  /usr/local/bin/gemma -g /home/rhea/account_wflow/twas_results/assoc_gemma_input/clop_noac_CrCl_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
  -notsnp \
  -p /home/rhea/account_wflow/phenos/clop_noac/clop_noac_rn_pheno_no_ids.txt \
  -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
  -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
  -lmm 4 \
  -o clop_noac_CrCl_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05
done


#new models
declare -a drugs=("clop_noac")
declare -a models=("ALL" "AFA" "CHN" "HIS")

#Use loop to run PrediXcan step two through GEMMA (accounts for relatedness)
for drug in ${drugs[@]}
  do
  for model in ${models[@]}
    do
    /usr/local/bin/gemma -g /home/rhea/account_wflow/pwas_results/blacks/clop_noac/assoc_gemma_input/blacks_clop_noac_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05_gemma.txt \
    -notsnp \
    -p /home/rhea/account_wflow/phenos/clop_noac/clop_noac_rn_pheno_no_ids.txt \
    -k /home/rhea/account_wflow/covariates/clop_noac/blacks_clop_noac.cXX.txt \
    -c /home/rhea/account_wflow/covariates/clop_noac/clop_noac_covariates.txt \
    -lmm 4 \
    -o ${drug}_PCAIR_filtered_${model}_baseline_rho0.1_zpval0.05
  done
done
```

## Correct Bonferroni

```{}
#Import necessary libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in drug name
"%&%" = function(a,b) paste(a,b,sep="")

#Create list for file input variants
model_list <- c("ALL", "AFA") 

#Make a data frame from combined PrediXcan results that includes chr #, bp, and BH and Bonferroni adj p-value
  #Read in output files
  #Create vector of p-values
  #Adjust p-values with BH and bonferroni methods
  #Add new columns with adjusted p-values to data frame
  #Join PrediXcan data frame with bp_chrome data frame to add columns with chr # and bp
  #Remove old/inaccurate chr and bp columns
  #Ouput data frame into directory

for (model in model_list) {
  assoc_file <- fread("/home/rhea/account_wflow/twas_results/assoc_gemma_output/clop_noac_CrCl_PCAIR_filtered_" %&% model %&% "_baseline_rho0.1_zpval0.05.assoc.txt")
  pvalues <- select(assoc_file, contains("p_wald"))
  pvalues <- as.vector(unlist(pvalues))
  pvalues_adjusted_BH <- p.adjust(pvalues, method = "BH")
  pvalues_adjusted_BF <- p.adjust(pvalues, method = "bonferroni")
  assoc_file <- add_column(assoc_file,  pvalues_adjusted_BH = pvalues_adjusted_BH , .before = "p_lrt")
  assoc_file <- add_column(assoc_file,  pvalues_adjusted_bonferroni = pvalues_adjusted_BF, .before = "p_lrt")
  fwrite(assoc_file, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_PCAIR_filtered_" %&% model %&% "_baseline_rho0.1_zpval0.05.adj.txt", sep = "\t", quote = F)
}
```

### Combine Significant Hits
```{}
#Import necessasry libraries
library(data.table)
library(dplyr)
library(tibble)

#Create function to paste in model name
"%&%" = function(a,b) paste(a,b,sep="")

#Create list of models for file input
model_list <- c("ALL", "AFA")

#Make a data frame with all results from each model per drug
  #Read in file
  #Add column containing model name
  #Compile significant subsets into single data frame
for(model in model_list){
  output <- fread("/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_PCAIR_filtered_" %&% model %&% "_baseline_rho0.1_zpval0.05.adj.txt")
  output <- add_column(output, model = model, .before = "chr")
  output <- add_column(output, drug = drug, .before = "chr")
  if(exists("all_assoc")){
      all_assoc<-left_join(x = all_assoc, y = output, all = TRUE)
  }
  else{
    all_assoc<-output
  }

  #subset at different thresholds
  all_sig<-subset(all_assoc, pvalues_adjusted_BH < .75)
  most_sig<-subset(all_assoc, pvalues_adjusted_BH < .1)
  BF_not1<-subset(all_assoc, pvalues_adjusted_bonferroni < 1)
  BF_sig<-subset(all_assoc, pvalues_adjusted_bonferroni < .05)
   
  #Output data frames into directory
  fwrite(all_assoc, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_allassoc_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt", na = "NA", quote = F, sep = "\t", col.names = T) 
  fwrite(all_sig, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt", na = "NA", quote = F, sep = "\t", col.names = T) 
  fwrite(most_sig, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_most_sig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt", na = "NA", quote = F, sep = "\t", col.names = T) 
  fwrite(BF_not1, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_BFnot1_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt", na = "NA", quote = F, sep = "\t", col.names = T) 
  fwrite(BF_sig, "/home/rhea/account_wflow/twas_results/adj_assoc_output/clop_noac_CrCl_BFsig_PCAIR_PAV_filtered_baseline_rho0.1_zpval0.05.adj.txt", na = "NA", quote = F, sep = "\t", col.names = T) 
}
```

No significant genes were found in this analysis. 

